lib/developer_portal/app/controllers/developer_portal/admin/account/personal_details_controller.rb

```diff
@@ -61,4 +58,16 @@ def redirect_path
     end
   end
 
+  def user_params
+    # There are extra fields customized by the provider so we can't add a permit here
+    params.require(:user)
+  end
+
+  def verify_current_password
+    return unless current_user.using_password?
+    return if current_user.authenticated?(user_params[:current_password])
+
+    flash.now[:error] = 'Current password is incorrect.'
+    resource.errors.add(:current_password, 'Current password is incorrect.')
```

Comment:
Maybe move that error to i18n locale? 
But just a suggestion.

Reply:
Assuming `resource` is an instance of `User` you can simply add an entry here:
https://github.com/3scale/porta/blob/master/config/locales/en.yml#L976-L979

With a keyword `invalid` I guess, like it's done here: https://github.com/3scale/porta/pull/3533/files#diff-44438ce218f5287c58d0017f965d888715635d94280669896f75841fbd7b4cd7R922

You will need to run this instead:
```suggestion
    resource.errors.add(:current_password, :invalid)
```

See [ActiveModel/Errors.html](https://api.rubyonrails.org/classes/ActiveModel/Errors.html#method-i-add) for reference.

Reply:
@josemigallas I've been playing around with this, but it doesn't work like this in this specific case.
The reason is that `:current_password` is not actually a real ActiveRecord attribute on the User model. Instead, this name (`current_password`) is used on the form - both for input, and to visualize the errors.

I will add an alternative suggestion :wink: